apps/molgenis-components/src/components/forms/EditModal.vue

```diff
@@ -0,0 +1,373 @@
+            v-if="showColumn(column)"
+            v-model="values[column.id]"
+            :columnType="column.columnType"
+            :description="column.description"
+            :errorMessage="errorPerColumn[column.id]"
+            :graphqlURL="graphqlURL"
+            :label="column.name"
+            :pkey="getPrimaryKey(values)"
+            :readonly="column.readonly || (pkey && column.key == 1 && !clone)"
+            :refBack="column.refBack"
+            :refLabel="column.refLabel"
+            :required="column.required"
+            :tableName="column.refTable"
+          />
+        </span>
+      </LayoutForm>
+    </template>
+    <template v-slot:footer>
+      <MessageSuccess v-if="success">{{ success }}</MessageSuccess>
+      <MessageError v-if="graphqlError">{{ graphqlError }}</MessageError>
+      <ButtonAlt @click="$emit('closeFormModal')">Close</ButtonAlt>
+      <ButtonOutline @click="saveDraft">Save draft</ButtonOutline>
+      <ButtonAction @click="save">Save {{ tableName }}</ButtonAction>
+    </template>
+  </LayoutModal>
+</template>
+
+<script>
+import Client from "../../client/client.js";
+import LayoutForm from "../layout/LayoutForm.vue";
+import LayoutModal from "../layout/LayoutModal.vue";
+import MessageError from "./MessageError.vue";
+import MessageSuccess from "./MessageSuccess.vue";
+import ButtonAction from "./ButtonAction.vue";
+import ButtonAlt from "./ButtonAlt.vue";
+import ButtonOutline from "./ButtonOutline.vue";
+import FormInput from "./FormInput.vue";
+import Expressions from "@molgenis/expressions";
+import constants from "../constants";
+import { getPrimaryKey } from "../utils";
+
+const { EMAIL_REGEX, HYPERLINK_REGEX } = constants;
+
+/**
+ * Properties:
+ * pkey:  when updating existing record, this is the primary key value
+ * clone: when you want to clone instead of update
+ * visibleColumns:  visible columns, useful if you only want to allow partial edit (array of strings)
+ * defaultValue: when creating new record, this is initialization value
+ * **/
+
+export default {
+  name: "EditModal",
+  data: function () {
+    return {
+      values: {},
+      errorPerColumn: {},
+      success: null,
+      tableMetaData: null,
+      data: null,
+      graphqlError: null,
+      client: null,
+    };
+  },
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+    pkey: Object,
+    clone: Boolean,
+    visibleColumns: Array,
+    defaultValue: Object,
+    graphqlURL: {
+      default: "graphql",
+      type: String,
+    },
+    tableName: {
+      type: String,
+      required: true,
+    },
+  },
+  components: {
+    LayoutForm,
+    FormInput,
+    ButtonAction,
+    ButtonAlt,
+    LayoutModal,
+    MessageError,
+    MessageSuccess,
+    ButtonOutline,
+  },
+  methods: {
+    saveDraft() {
+      this.save(true);
+    },
+    async save(isDraft) {
+      let promise;
+      if (this.pkey && !this.clone) {
+        promise = this.client.updateVariables(
```

Comment:
maybe do `
const saveMethod = this.pkey && !this.clone ? 'updateVariables' : 'insertVariable'
await this.client[saveMethod](params)
`

or some type of apply ?